azure-iot-device/azure/iot/device/provisioning/pipeline/pipeline_ops_provisioning.py

```diff
@@ -99,21 +99,21 @@ class SendRegistrationRequestOperation(PipelineOperation):
     This operation is in the group of DPS operations because it is very specific to the DPS client.
     """
 
-    def __init__(self, request_id, request_payload, registration_id, callback=None):
+    def __init__(self, request_payload, registration_id, registration_result=None, callback=None):
         """
         Initializer for SendRegistrationRequestOperation objects.
 
-        :param request_id : The id of the request being sent
         :param request_payload: The request that we are sending to the service
         :param registration_id: The registration ID is used to uniquely identify a device in the Device Provisioning Service.
         :param Function callback: The function that gets called when this operation is complete or has failed.
          The callback function must accept A PipelineOperation object which indicates the specific operation which
          has completed or failed.
         """
         super(SendRegistrationRequestOperation, self).__init__(callback=callback)
-        self.request_id = request_id
         self.request_payload = request_payload
         self.registration_id = registration_id
+        self.registration_result = registration_result
+        self.retry_after_timer = None
 
 
 class SendQueryRequestOperation(PipelineOperation):
```

Comment:
>SendQueryRequestOperation [](start = 6, length = 25)

Likewise, this isn't just sending a request. It also waits for the response. Which is to say, it's the entire query process. Perhaps `QueryRegistrationStatusOperation`?

Reply:
I kept the name sort of similar to the registration operation. Both starting with send and both having the word request.  if you have any suggestion for the registration as well , we can consider renaming both.

---
In reply to: [365414678](https://github.com/Azure/azure-iot-sdk-python/pull/432#discussion_r365414678) [](ancestors = 365414678)

Reply:
I agree that the other also needs to be renamed - I thought I had left a comment about that as well. If you're sending a request and waiting for a response, it is not merely sending a request, and thus needs to be renamed to the high level operation itself.

---
In reply to: [365650046](https://github.com/Azure/azure-iot-sdk-python/pull/432#discussion_r365650046) [](ancestors = 365650046,365414678)

Reply:
commented on the other one. 

---
In reply to: [366018370](https://github.com/Azure/azure-iot-sdk-python/pull/432#discussion_r366018370) [](ancestors = 366018370,365650046,365414678)